### PR TITLE
ADM-708:[backend]feat: add get board info api

### PR DESCRIPTION
**Pull request by weiraneve**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by BoBoDai — APPROVED** [review]

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 1]
```diff
@@ -132,6 +128,46 @@ public JiraVerifyDTO verify(BoardType boardType, BoardVerifyRequestParam boardVe
...
+	public BoardConfigDTO getInfo(BoardType boardType, BoardRequestParam boardRequestParam) {
+		URI baseUrl = urlGenerator.getUri(boardRequestParam.getSite());
+		try {
+			if (!BoardType.JIRA.equals(boardType)) {
+				throw new BadRequestException("boardType param is not correct");
+			}
+			String jiraBoardStyle = jiraFeignClient
+				.getProject(baseUrl, boardRequestParam.getProjectKey(), boardRequestParam.getToken())
+				.getStyle();
+			BoardType jiraBoardType = "classic".equals(jiraBoardStyle) ? BoardType.CLASSIC_JIRA : BoardType.JIRA;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:140]: `"classic"` should be a value of enum, not customized

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 2]
```diff
@@ -132,6 +128,46 @@ public JiraVerifyDTO verify(BoardType boardType, BoardVerifyRequestParam boardVe
...
+							.targetFields(neededTargetFields)
+							.jiraColumnResponse(jiraColumnResult.getJiraColumnResponse())
+							.ignoredTargetFields(ignoredTargetFields)
+							.users(users)
+							.build())
+				.join();
+		}
+		catch (RuntimeException e) {
+			Throwable cause = Optional.ofNullable(e.getCause()).orElse(e);
+			log.error("Failed when call Jira to get board config, project key: {}, board id: {}, e: {}",
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:161]: Failed to call ...
not Failed when ....

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 3]
```diff
@@ -132,6 +128,46 @@ public JiraVerifyDTO verify(BoardType boardType, BoardVerifyRequestParam boardVe
...
+					boardRequestParam.getBoardId(), boardRequestParam.getProjectKey(), cause.getMessage());
+			if (cause instanceof BaseException baseException) {
+				throw baseException;
+			}
+			throw new InternalServerErrorException(
+					String.format("Failed when call Jira to get board config, cause is %s", cause.getMessage()));
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:167]: same as previous

**Review by guzhongren — COMMENTED** [review]
refactor

**Comment by guzhongren** [comment]
@weiraneve

**Pull request merged**